server/src/test/java/io/spine/examples/chatspn/server/ChatSpnServerTest.java

```diff
@@ -0,0 +1,210 @@
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.chatspn.server;
+
+import io.grpc.ManagedChannel;
+import io.spine.base.EventMessageField;
+import io.spine.base.Field;
+import io.spine.client.Client;
+import io.spine.client.EventFilter;
+import io.spine.examples.chatspn.account.UserProfile;
+import io.spine.examples.chatspn.account.event.AccountCreated;
+import io.spine.server.Server;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import static io.grpc.ManagedChannelBuilder.forAddress;
+import static io.spine.client.Client.usingChannel;
+import static io.spine.client.ConnectionConstants.DEFAULT_CLIENT_SERVICE_PORT;
+import static io.spine.examples.chatspn.server.ExpectedOnlyAssertions.assertExpectedFields;
+import static io.spine.examples.chatspn.server.given.ChatSpnServerTestEnv.accountCreated;
+import static io.spine.examples.chatspn.server.given.ChatSpnServerTestEnv.createAccount;
+import static io.spine.examples.chatspn.server.given.ChatSpnServerTestEnv.userProfile;
+import static io.spine.util.Exceptions.newIllegalStateException;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+@DisplayName("ChatSPN server should")
+final class ChatSpnServerTest {
+
+    private static final String ADDRESS = "localhost";
+
+    private final Collection<ManagedChannel> channels = new ArrayList<>();
+    private final Collection<Client> clients = new ArrayList<>();
+    private Server server;
+
+    /**
+     * Starts the server before each test case.
+     */
+    @BeforeEach
+    void startServer() throws IOException {
+        server = ChatSpnServer.create();
+        server.start();
+    }
+
+    /**
+     * Shutdowns the server after each test case.
+     */
+    @AfterEach
+    void stopServer() {
+        clients.forEach(Client::close);
+        channels.forEach(channel -> {
+            channel.shutdown();
+            try {
+                channel.awaitTermination(1, SECONDS);
+            } catch (InterruptedException e) {
+                throw newIllegalStateException(e, "Channel can't be terminated");
+            }
+        });
+        server.shutdown();
+    }
+
+    @Test
+    @DisplayName("handle command and request for projection status")
```

Comment:
"Call a method and then call another method, receiving some Proto message is response"?